### ISSUE TITLE
iCade and 4-directional joystick behavior request

### ISSUE THREAD

**Issue by GoogleCodeExporter** [mrpostiga/imame4all#93]
```
I have a suggestion that I think would improve 4-way diagonal handling 
specifically for 4-way games on the iCade.

The current behavior is that in 4 way mode, if you roll the stick from a single 
axis to the diagonal, you'll stop moving - I'm assuming it's forcing the 
equivalent of a key-up for all directions that exist on the diagonal axis, 
since the diagonal is 'disabled'.

I think a more desirable and true-to-arcade behavior would be to retain the 
activated state of the last axis, while ignoring the new axis - 

i.e. -
If I am moving left, and then push the stick to up-left, the left axis would 
remain depressed, but the up axis would be ignored. Rolling fully to the up 
state would release the left axis, as expected, and engage the up axis.  If I 
were to repeat the process in reverse, moving upward, and then rolling the 
stick to up-left, I'd retain the up-axis activation, but the left axis would be 
ignored. 

Basically, when moving into a diagonal state, retain any previously activated 
axis while ignoring any new ones.

Does that make any sense? It would resolve the fairly constant issues with 
pressing into the diagonal and losing both axes. I'm not sure if this is easily 
resolvable with the current input-handling code ( presumably in four-way mode, 
if one of the axes of a diagonal is depressed, you'd ignore any keydown events 
for the paired axis )


```

Original issue reported on code.google.com by `travisba...@gmail.com` on 25 Dec 2011 at 10:19
